uiowa/uiowa
# [Events] Exposed filters for local event list block

Resolves https://github.com/uiowa/uiowa/issues/8041. 




# How to test

- Code review
  - [What's the best way to set the block date to today](https://github.com/uiowa/uiowa/blob/f2bec976846c65aae7904516e6d5067db1826a86/docroot/modules/custom/sitenow_events/sitenow_events.module#L128)? 

```
ddev composer install && ddev blt frontend && ddev blt ds --site=iowasummerwritingfestival.uiowa.edu && ddev drush @iowasummerwritingfestival.local uli  /page-list-block-test
```
1. Edit the block
2. Scroll to the filters section and check each filter to expose them to users
3. Save block and test the filters
4. Confirm that the block date is set to today
5. Confirm that exposed filters work as expected
6. Test `/events` page and confirm start and end date fields are working

## Test other views

```
ddev blt ds --site=oniowa.uiowa.edu && ddev drush @oniowa.local uli  /events
```
1. Test OnIowa date filter
2. Submit and compare against prod

```
ddev blt ds --site=iisc.uiowa.edu && ddev drush @oniowa.local uli  /projects
```

```
ddev blt ds --site=classrooms.uiowa.edu && ddev drush @oniowa.local uli  /spaces/find-university-classroom
```
1. Also test `/spaces/programmed-classrooms`

```
ddev blt ds --site=commencement.uiowa.edu && ddev drush @oniowa.local uli  /ceremonies
```
```
ddev blt ds --site=inrc.law.uiowa.edu && ddev drush @lawinrc.local uli  /iowa-grants-guide
```
```
ddev drush @default.local sql-drop -y && ddev drush @default.local cr && ddev blt di --site=default && ddev drush @default.local config-split:activate event && ddev drush @default.local uli admin/config/sitenow/sitenow-events
```
